Backport I²C to renesas 2.5 branch

aaillet opened the pull request:
Backport :
- Revert new clock controller mechanism used in current renesas-next
- Delete the adition of "Kconfig.test" in drivers/i2c/Kconfig

aaillet committed:
dts: bindings: i2c: add definition for Renesas RCar I2C

Signed-off-by: Aymeric Aillet <aymeric.aillet@iot.bzh>

aaillet committed:
drivers: i2c: Driver enhancements

Functions reordering
Change returns value to match i2c api
Update Kconfig in order to compile the driver
Rename i2c_finish function
Revert changes for i2c_transfer returns values.

Signed-off-by: Aymeric Aillet <aymeric.aillet@iot.bzh>

aaillet committed:
drivers: i2c: add Renesas RCar I2C

based on uboot rcar i2c driver
I2C add bus clk management
also fix uart clock definition
set_speed impl inspired from uboot driver
set addr + transfer to do
Some fix on i2c config & driver for rcar gen3
- icccr is now local
- use busy wait to wait the right amount of time (in theory),
the current time used isn't as precise as needed
- fix call for set addr that was closing with error in the wrong case
- now call for set addr for each message
- fix write and read common fonction name
- Delete the function i2c_rcar_set_speed
- Set the ICCCR reg following documentation recommendation
- Set ICCCR reg in i2c_rcar_configure
- rename defines
- restructure if statements
- rework comments
- refactor write_common and read_common functions

tested with either 100hz & 400hz on i2c2 bus

Signed-off-by: Aymeric Aillet <aymeric.aillet@iot.bzh>

# Conflicts:
#	drivers/i2c/Kconfig

julien-massot commented on drivers/i2c/i2c_rcar.c:
```diff
@@ -0,0 +1,353 @@
+
+struct i2c_rcar_cfg {
+	uint32_t reg_addr;
+	int reg_size;
```
unused

julien-massot commented on drivers/clock_control/clock_control_rcar_mssr.c:
```diff
@@ -48,7 +48,8 @@ static const uint16_t srcr[] = {
 #define CANFDCKCR_DIVIDER_MASK 0x1FF
 
-#define S3D4_CLK_RATE 66600000
+#define S3D4_CLK_RATE 66660000	/* 66.66 MHz */
```
looks unrelated to this pr

julien-massot commented on drivers/i2c/i2c_rcar.c:
```diff
@@ -0,0 +1,353 @@
+#define RCAR_I2C_ICMCR_TSBE		BIT(2)
+#define RCAR_I2C_ICMCR_FSB		BIT(1) /* force stop bit */
+#define RCAR_I2C_ICMCR_ESG		BIT(0) /* enable start bit gen */
+#define RCAR_I2C_ICSSR			0x08 /* slave status */
```
indent is odd here

julien-massot commented on drivers/i2c/i2c_rcar.c:
```diff
@@ -0,0 +1,353 @@
+
+	do {
+		/* Send address after any Start condition */
+		if (i2c_rcar_set_addr(dev, addr, !!(msgs->flags & I2C_MSG_READ))) return -EIO; /* No ACK received */
```
please move return statement to the next line and add brace

julien-massot commented on drivers/i2c/i2c_rcar.c:
```diff
@@ -0,0 +1,353 @@
+
+		/* Transfer data */
+		if (msgs->len) {
+			i2c_rcar_transfer_msg(dev, msgs);
```
you do not check return of this function, we may stop sending msg if one failed ?

julien-massot commented on drivers/i2c/i2c_rcar.c:
```diff
@@ -0,0 +1,353 @@
+		break;
+	default:
+		return -ENOTSUP;
+	}
```
missing break there for default case

julien-massot replied:
hum break after a return is not so usefull anyway :)

julien-massot commented on drivers/i2c/i2c_rcar.c:
```diff
@@ -0,0 +1,353 @@
+				(clock_control_subsys_t *) &config->bus_clk,
+				&data->clk_rate);
+		if (ret < 0) return ret;
+	}
```
please put return on a new line and add brace

aaillet committed:
drivers: i2c: Fix driver compatibility

The driver is not only ocmpatible with R-Car Gen3 board
but with any R-Car generation.

Signed-off-by: Aymeric Aillet <aymeric.aillet@iot.bzh>

# Conflicts:
#	dts/arm/renesas/gen3/rcar_gen3_cr7.dtsi

aaillet committed:
doc: Add I2C documentation for rcar_ulcb

Add documentation regarding I2C driver for Renesas rcar_ulcb boards.

Signed-off-by: Aymeric Aillet <aymeric.aillet@iot.bzh>
